packages/yew/src/html/classes.rs

```diff
@@ -16,8 +15,28 @@ pub struct Classes {
     set: IndexSet<Cow<'static, str>>,
 }
 
+/// helper method to efficiently turn a set of classes into a space-separated
+/// string. Abstracts differences between ToString and IntoPropValue. The
+/// `rest` iterator is cloned to pre-compute the length of the String; it
+/// should be cheap to clone.
+fn build_string<'a>(first: &'a str, rest: impl Iterator<Item = &'a str> + Clone) -> String {
+    // The length of the string is known to be the length of all the
+    // components, plus one space for each element in `rest`.
+    let mut s = String::with_capacity(
+        rest.clone()
```

Comment:
This makes sense mostly in the case where the same classes object is being repeatedly being converted to a string. If the average classes object only gets converted into a string once (for instance, every time `Component::yew` is called), then you probably won't notice any speedups, because the summation happens only once either way (and doesn't have to pay duplicate detection penalties if we track it per-insert)